asset/filtered_manager.go

```diff
@@ -0,0 +1,65 @@
+package asset
+
+import (
+	"github.com/sensu/sensu-go/types"
+	"github.com/sensu/sensu-go/util/eval"
+)
+
+// NewFilteredManager returns an asset Getter that filters assets based on the
+// given entity. Assets that aren't filtered get passed to the underlying
+// getter, allowing composition with other asset managers.
+func NewFilteredManager(getter Getter, entity *types.Entity) *filteredManager {
+	return &filteredManager{
+		getter: getter,
+		entity: entity,
+	}
+}
+
+// filteredManager evaluates asset filter(s) and calls the underlying
+// asset Getter based on filter result.
+type filteredManager struct {
+	entity *types.Entity
+	getter Getter
+}
+
+// Get fetches, verifies, and expands an asset, but only if it is not
+// filtered.
+func (f *filteredManager) Get(asset *types.Asset) (*RuntimeAsset, error) {
+	filtered, err := f.isFiltered(asset)
+	if err != nil {
+		return nil, err
+	}
+
+	if filtered {
+		return nil, nil
+	} else {
+		return f.getter.Get(asset)
+	}
+}
+
+// isFiltered evaluates the given asset's filters and returns true if any of
+// them match the current entity.
+func (f *filteredManager) isFiltered(asset *types.Asset) (bool, error) {
```

Comment:
If multiple filters exist, they are to be AND'd, but this ORs them.

Reply:
```
// isFiltered... returns true if any of them match the current entity
```
description and functionality should actually reflect
```
// isFiltered... returns true if all of them match the current entity
```
Let me update the function and the tests here.